src/wp-includes/media.php

```diff
@@ -5455,6 +5459,21 @@ function wp_get_loading_attr_default( $context ) {
 		return false;
 	}
 
+	// Conditionally skip lazy-loading on images before the loop.
+	if ( 'wp_get_attachment_image' === $context || 'the_post_thumbnail' === $context ) {
+		// Only apply for main query but before the loop.
+		if ( $wp_query->before_loop && is_main_query() ) {
+			/*
+			 * Any image before the loop, but after the header has started should not be lazy-loaded,
+			 * except when the footer has already started which can happen when the current template
+			 * does not include any loop.
+			 */
+			if ( did_action( 'get_header' ) && ! did_action( 'get_footer' ) ) {
+				return false;
+			}
+		}
+	}
```

Comment:
Why is there three nested if statements here? Wouldn't one if statement do the same?

Reply:
Either works, I would agree with you that putting all this in one if statement makes more sense from a code perspective. We should wrap it in multiple lines though so that we can keep the comments that explain all of this, as it's fairly complex.
```suggestion
	if (
		( 'wp_get_attachment_image' === $context || 'the_post_thumbnail' === $context )
		// Only apply for main query but before the loop.
		&& $wp_query->before_loop && is_main_query()
		/*
		 * Any image before the loop, but after the header has started should not be lazy-loaded,
		 * except when the footer has already started which can happen when the current template
		 * does not include any loop.
		 */
		&& did_action( 'get_header' ) && ! did_action( 'get_footer' )
	) {
		return false;
	}
```

Reply:
Please change to one if statement.

Reply:
Updated.